src/handler/send_token_data.c

```diff
@@ -0,0 +1,33 @@
+#include "send_token_data.h"
+
+#include "token_parser.h"
+#include "signature.h"
+#include "../globals.h"
+#include "types.h"
+#include "../sw.h"
+
+int handler_send_token_data(bool first, buffer_t *cdata) {
+    token_t token = {0};
+
+    if (first) {
+        explicit_bzero(&G_context, sizeof(G_context));
+        explicit_bzero(&G_token_symbols, sizeof(G_token_symbols));
+    }
+
+    if (G_token_symbols.offset >= TX_MAX_TOKENS) {
+        // too many tokens
+        explicit_bzero(&G_token_symbols, sizeof(G_token_symbols));
+        return io_send_sw(SW_WRONG_DATA_LENGTH);
+    }
+
+    if (!check_token_signature_from_apdu(cdata, &token)) return io_send_sw(SW_INVALID_SIGNATURE);
+
+    // save token/symbol on global ctx
+    token_symbol_t *token_symbol = &G_token_symbols.tokens[G_token_symbols.offset++];
```

Comment:
Just a question.

We are saving the tokens data on this global variable `G_token_symbols`, but we are not reading this data anywhere yet, right?

What's the next step? We will have another handler to send a tx using the tokens?

Reply:
This global context will be maintained between commands, the sign_tx command will check that all token uids of the transaction are on `G_token_symbols`, meaning they were verified already and will use the symbol to show the value of the outputs.